### PR TITLE
bpo-30532: email.policy.SMTP.fold() mangles long headers

### PR DESCRIPTION
**Pull request by jhillacre**
Pull request for http://bugs.python.org/issue30532

### CODE REVIEW

File: Lib/email/_header_value_parser.py [hunk 1]
```diff
@@ -343,7 +343,7 @@ def _fold(self, folded):
             if ws is not None:
                 # Peel off the leading whitespace and make it sticky, to
                 # avoid infinite recursion.
```
Comment (zvyn) [inline Lib/email/_header_value_parser.py]: Looks good to me! You could remove the repeated comment from line 340 while you are here.

**Comment by jhillacre** [comment]
@zvyn OK, got it in 3d07dc5.

**Comment by jhillacre** [comment]
Rebased from master so the windows build passes and gives the PR the ✔️.

**Comment by jhillacre** [comment]
Is there anything I need to do to keep the PR moving?

File: Lib/email/_header_value_parser.py [hunk 2]
```diff
@@ -341,9 +341,7 @@ def _fold(self, folded):
             # avoid infinite recursion.
             ws = part.pop_leading_fws()
             if ws is not None:
-                # Peel off the leading whitespace and make it sticky, to
-                # avoid infinite recursion.
-                folded.stickyspace = str(part.pop(0))
+                folded.stickyspace = ws
```
Comment (bitdancer) [inline Lib/email/_header_value_parser.py]: It looks like this was a case of me "fixing" the code, and missing one spot (for which there obviously wasn't a test).  To make this code match other other instances of this logic, you should put an str call around ws.

**Review by bitdancer — CHANGES_REQUESTED** [review]
One small change, and we need a NEWS entry (which should now use the 'blurb' mechanism, see the recent updates to the devguide).

**Comment by jhillacre** [comment]
@bitdancer, Added your requested changes in ec63c3c and 29f92f0.

Should I be rebasing this pull request to a single commit with the message of `bpo-30532: Fix email header value parser dropping folding white space in certain cases.`?

**Review by bitdancer — APPROVED** [review]

**Comment by bitdancer** [comment]
Nope, it gets squashed during merge.  Now it needs backport to 3.6, and I think 3.5 is still open for backports.  You can do that (see devguide), or someone else will, since it isn't (yet) automated.

**Comment by jhillacre** [comment]
added backport PRs #2591 #2592

**Pull request merged**